templates/aws-stack.yml

```diff
@@ -665,6 +666,16 @@ Resources:
       SubnetId: !Ref Subnet1
       RouteTableId: !Ref Routes
 
+  # A resource that depends on the leaf nodes of the VPC configuration so that
+  # a strict ordering can be established on teardown.
+  VpcComplete:
+    Type: AWS::CloudFormation::WaitConditionHandle
+    Metadata:
+      VpcResources: !If
+        - CreateVpcResources
+        - [ !Ref RouteDefault, !Ref Subnet0Routes, !Ref Subnet1Routes ]
```

Comment:
I assume these are meaningless as metadata, and only present to make this `VpcComplete` resource implicitly depend on those other resources? And maybe you can't use a regular `DependsOn` because they don't support conditionality like `!If CreateVpcResources`?

Reply:
Exactly correct :+1: